### PR TITLE
 refactor(store): use react-query for bns name requests

### LINKED ISSUE #1801: Remove async atom `namesResponseState`

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1333880154).

- Closes #1801
- Updates react query
- Creates `query` folder for ReactQuery hooks, file name convention `*.query.ts`

### CODE REVIEW

**Review by beguene — APPROVED** [review]
Looks good ! We should merge all our refactor PRs as quickly as possible.
@kyranjamie @fbwoolf @andresgalante Do we need QA for each PRs or is it not better this time that we merge everything then do QA regression at the end ?

**Review by fbwoolf — APPROVED** [review]
Nice refactor!

**Comment by timstackblock** [comment]
> Looks good ! We should merge all our refactor PRs as quickly as possible. @kyranjamie @fbwoolf @andresgalante Do we need QA for each PRs or is it not better this time that we merge everything then do QA regression at the end ?

We can do a smoke test of the changes on each PR and then do a full regression on the final PR. In order to make this work we need to change the final PR to have a downloadable "stacks-wallet-chromium" file. It currently is not generated the same way as the individual branch PRs.

**Comment by fbwoolf** [comment]
> Looks good ! We should merge all our refactor PRs as quickly as possible. @kyranjamie @fbwoolf @andresgalante Do we need QA for each PRs or is it not better this time that we merge everything then do QA regression at the end ?

I think fine if devs test and merge into one to test by QA altogether.

File: src/store/common/api-clients.hooks.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { useAtomValue } from 'jotai/utils';
 import { apiClientState } from '@store/common/api-clients';
 
-export function useAccountsApi() {
+export function useApi() {
   return useAtomValue(apiClientState);
 }
+
+export type Api = ReturnType<typeof useApi>;
```
Comment (fbwoolf) [inline src/store/common/api-clients.hooks.ts:8]: This is a nice idea ...branched off your changes and using it.

File: src/query/bns/bns.query.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { useQueries, useQuery, UseQueryResult } from 'react-query';
+
+import { useApi, Api } from '@store/common/api-clients.hooks';
+import { BnsNamesOwnByAddressResponse } from '@stacks/blockchain-api-client';
+
+const STALE_TIME = 15 * 60 * 1000; // 15 min
```
Comment (fbwoolf) [inline src/query/bns/bns.query.ts:6]: Should this be available globally?

**Pull request merged**